include/utils.h

```diff
@@ -207,7 +207,8 @@ template <std::integral Integral>
 }
 
 template <typename Void>
-requires(std::is_void_v<Void>) [[nodiscard]] constexpr Void* offset(Void* const ptr, const size_t offset_bytes) {
+    requires(std::is_void_v<Void>)
```

Comment:
I guess this is one of the things we could make a decision on, between the C++20 switch and the new clang format version.

I am happy with this default, FWIW.

Reply:
+1 new default, these are easy to miss if not on a separate line.